[DOCS-7077] Update and reorg main SDS doc (#21353)
DataDog/documentation: content/en/sensitive_data_scanner/_index.md

- [Using the out-of-the-box dashboard](#out-of-the-box-dashboard).

{{< img src="sensitive_data_scanner/sds_main_12_01_24.png" alt="The Sensitive Data Scanner page showing six out of the 12 active scanning groups" style="width:90%;">}}

## Permissions

By default, users with the Datadog Admin role have access to view and set up scanning rules. To allow other users access, grant the `data_scanner_read` or `data_scanner_write` permissions under [Compliance][1] to a custom role. See [Access Control][2] for details on how to set up roles and permissions.

{{< img src="sensitive_data_scanner/read_write_permissions.png" alt="The compliance permissions sections showing data scanner read and writer permissions" style="width:55%;">}}

## Set up Sensitive Data Scanner

This section walks you through the following to set up Sensitive Data Scanner:

- [Adding a scanning group](#add-a-scanning-group)
- [Adding scanning rules](#add-scanning-rules)
- [Controlling access to events with sensitive data](#control-access-to-events-with-sensitive-data)

You can also [redact sensitive data in tags](#redact-sensitive-data-in-tags).

### Add a scanning group

> A scanning group determines what data to scan. It consists of a query filter and a set of toggles to enable scanning for Logs, APM, RUM, and Events. See the [Log Search Syntax][3] documentation to learn more about query filters.

Reviewer: ```suggestion
A scanning group determines what data to scan. It consists of a query filter and a set of toggles to enable scanning for logs, APM, RUM, and events. See the [Log Search Syntax][3] documentation to learn more about query filters.
```

> For Terraform, see the [Datadog sensitive data scanner group][4] resource.

Reviewer: ```suggestion
For Terraform, see the [Datadog Sensitive Data Scanner group][4] resource.
```

To set up a scanning group:

> 1. Navigate to [Sensitive Data Scanner][5]. Alternatively, go to Organization Settings > Sensitive Data Scanner.

Reviewer: ```suggestion
1. Navigate to [Sensitive Data Scanner][5]. Alternatively, go to **Organization Settings** > **Sensitive Data Scanner**.
```

1. Click **Add scanning group**. Alternatively, click the **Add** dropdown menu on the top right corner of the page and select **Add Scanning Group**.
1. Enter a query filter for the data you want to scan. At the top, click **APM Spans** to preview the filtered spans. Click **Logs** to see the filtered logs.
1. Enter a name and description for the group.

> 1. Click the toggle buttons to enable Sensitive Data Scanner for the products you want (for example, Logs, APM Spans, RUM Events, and Events).

Reviewer: ```suggestion
1. Click the toggle buttons to enable Sensitive Data Scanner for the products you want (for example, logs, APM spans, RUM events, and general events).
```

There may be a better way to say "general events", but it just stuck out to me to say "RUM events and events"

Author: Thanks, updated to `Datadog events`

1. Select whether you want to add a library rule or create a custom scanning rule.

{{< tabs >}}
{{% tab "Add scanning rule from the library rules" %}}

The Scanning Rule Library contains predefined rules for detecting common patterns such as email addresses, credit card numbers, API keys, authorization tokens, and more.

1. In the **Add library rules to the scanning group** section, select the library rules you want to use.
1. In the **Define rule target and action** section, select if you want to scan the **Entire Event** or **Specific Attributes**.
    - If you are scanning the entire event, you can optionally exclude specific attributes from getting scanned.
    - If you are scanning specific attributes, specify which attributes you want to scan.
{{% sds-scanning-rule %}}
1. Click **Add Rules**.

{{% /tab %}}
{{% tab "Add a custom scanning rule" %}}

You can create custom scanning rules using regex patterns to scan for sensitive data.

1. In the **Define match conditions** section, specify the regex pattern to use for matching against events in the **Define regex** field. Enter sample data in the **Regex tester** field to verify that your regex pattern is valid.   

>     Sensitive Data Scanner supports Perl Compatible RegEx (PCRE), but the following patterns are not supported:

Reviewer: ```suggestion
    Sensitive Data Scanner supports Perl Compatible Regular Expressions (PCRE), but the following patterns are not supported:
```

    - Backreferences and capturing sub-expressions (lookarounds)
    - Arbitrary zero-width assertions
    - Subroutine references and recursive patterns
    - Conditional patterns
    - Backtracking control verbs

>     - The \C "single-byte" directive (which breaks UTF-8 sequences)

Reviewer: ```suggestion
    - The `\C` "single-byte" directive (which breaks UTF-8 sequences)
```

>     - The \R newline match

Reviewer: ```suggestion
    - The `\R` newline match
```

>     - The \K start of match reset directive

Reviewer: ```suggestion
    - The `\K` start of match reset directive
```

    - Callouts and embedded code
    - Atomic grouping and possessive quantifiers
{{% sds-scanning-rule %}}
1. Click **Add Rule**.

{{% /tab %}}
{{< /tabs >}}

**Notes**:

- Any rules that you add or update only affect data coming into Datadog after the rule was defined.
- Sensitive Data Scanner does not affect any rules you define on the Datadog Agent directly.
- To turn off Sensitive Data Scanner entirely, set the toggle to **off** for each Scanning Group and Scanning Rule so that they are disabled.

> See [Investigate Sensitive Data Issues][7] for details on how to use the [Summary][8] page to triage your sensitive data issues.

Author: This link will work after [this PR](https://github.com/DataDog/documentation/pull/21567) is merged into the `may/main-sds-refresh` branch.
